cluster/provision_gateway.sh

```diff
@@ -0,0 +1,675 @@
+# https://github.com/Juniper/contrail-kubernetes
+#
+# Author - Sanju Abraham -@asanju- OpenContrail-Kubernetes
+#
+##############################################################
+set -x
+
+source /etc/contrail/opencontrail-rc
+
+readonly PROGNAME=$(basename "$0")
+
+runok="/etc/contrail/vrouter_sgw_prov_run.ok"
+ocver=$OPENCONTRAIL_TAG
+ockver=$OPENCONTRAIL_KUBERNETES_TAG
+OPENCONTRAIL_PUBLIC_SUBNET="${OPENCONTRAIL_PUBLIC_SUBNET:-10.1.0.0/16}"
+GCLOUD_NET_ADDR=""
+
+timestamp() {
+    date
+}
+
+if [ ! -f /var/log/contrail/provision_gateway.log ]; then
+   mkdir -p /var/log/contrail
+   touch /var/log/contrail/provision_gateway.log
+fi
+
+log_error_msg() {
+    msg=$1
+    echo "$(timestamp): ERROR: $msg"
+}
+
+log_warn_msg() {
+    msg=$1
+    echo "$(timestamp): WARNING: $msg"
+}
+
+log_info_msg() {
+    msg=$1
+    echo "$(timestamp): INFO: $msg"
+}
+
+log_info_msg "Start Provisioning VROUTER kernel module and agent in container"
+
+LOG_FILE=/var/log/contrail/provision_gateway.log
+exec 2>&1 &> >(tee -a "$LOG_FILE")
+
+REDHAT="redhat"
+UBUNTU="ubuntu"
+VROUTER="vrouter"
+VHOST="vhost0"
+
+if [[ -z $ocver ]]; then
+   ocver="R2.20"
+fi
+
+hname=`hostname`
+
+function detect_os()
+{
+   OS=`uname`
+   OS_TYPE="none"
+   if [ "${OS}" = "Linux" ]; then
+      if [ -f /etc/redhat-release ]; then
+         OS_TYPE="redhat"
+      elif [ -f /etc/debian_version ]; then
+         OS_TYPE="ubuntu"
+      fi
+   fi
+}
+
+function generate_rc()
+{
+flag=false
+rcfile="/etc/contrail/opencontrail-rc"
+rcdir=`dirname "$rcfile"`
+if [ ! -f "$rcdir" ]; then
+    mkdir -p "$rcdir"
+fi
+if [[ -z $OPENCONTRAIL_CONTROLLER_IP ]]; then
+    if isGceVM ; then
+       api_server=$(cat /etc/salt/minion.d/grains.conf | grep api_servers | awk '{print $2}' | tr -d "'")
+    fi
+
+   # Try to resolve
+   if [[ -n $api_server ]]; then
+       OPENCONTRAIL_CONTROLLER_IP=$(host $api_server | grep address | awk '{print $4}')
+   else
+      log_error_msg "Unable to find contrail controller"
+   fi
+   echo "OPENCONTRAIL_CONTROLLER_IP=$OPENCONTRAIL_CONTROLLER_IP" >> $rcfile
+fi
+if [[ -z $OPENCONTRAIL_VROUTER_INTF ]];then
+   OPENCONTRAIL_VROUTER_INTF="eth0"
+   echo "OPENCONTRAIL_VROUTER_INTF="eth0"" >> $rcfile
+   flag=true
+fi
+if [[ "$flag" == true ]]; then
+   source "$rcfile"
+fi
+
+MINION_OVERLAY_NET_IP=$(/sbin/ifconfig $OPENCONTRAIL_VROUTER_INTF | grep "inet addr" | awk -F: '{print $2}' | awk '{print $1}')
+if [ -z $MINION_OVERLAY_NET_IP ]; then
+   # Check if vhost is already up with the IP
+   MINION_OVERLAY_NET_IP=$(ip a | grep vhost0 | grep inet | awk -F/ '{print $1}' | awk '{print $2}')
+fi
+if [ -z $MINION_OVERLAY_NET_IP ]; then
+     msg="Unable to get an IP address on the $OPENCONTRAIL_VROUTER_INTF or vhost0 interface.
+         Please check the interface and IP address assignement and restart provision"
+    log_error_msg "$msg"
+    echo "$msg"
+    exit
+fi
+}
+
+function prep_to_install()
+{
+  if [ "$OS_TYPE" == $REDHAT ]; then
```

Comment:
This is installing a bunch of software into the host OS. Which is undesirable.
It should change to build the kernel module inside a docker container. The ansible scripts do that.